src/scripts/editor/editor.js

```diff
@@ -186,6 +186,32 @@ function generateColorWheel(width, height) {
     return rendering;
 }
 
+function filterJavascriptByPurposes(sourceCode, purposes) {
+    // Prepend "CODE_RUNTIME" so it's default.  Split code into blocks by "CODE_..." lines.
+    let codeBlocks = `//! CODE_RUNTIME\n${sourceCode}`.split(/^\/\/! *CODE_/m);
+    // Filter out any code blocks that don't match the given purposes.
+    const purposesRegex = new RegExp(`^(?:${purposes.join("|")})(?:\n|\r\n)`);
+    const result = codeBlocks.
+        filter(block => block.match(purposesRegex)).
+    // Remove the start line for each code block (the remains of the "//! CODE_" line).
+        map(block => block.slice(block.indexOf('\n')+1)).
+    // Rejoin modified blocks into a whole
+        join("");
+    return result;
+}
+
+function getRunnableJavascriptForOnePlugin(event, purposes) {
+    const configFields = event.fields.filter((field) => field.key !== "plugin");
+    const configsJS = `const CONFIG = { fields: ${JSON.stringify(configFields)} };`;
+    let pluginJS = FIELD(event, "plugin", "javascript");
+    pluginJS = filterJavascriptByPurposes(pluginJS, purposes);
+    pluginsJS = `// PLUGIN CODE"\n${pluginJS}\n`;
+    if (!pluginsJS.replace(/\/\/[^\n]*\n|[\n\t ]/g, "")) {
+        return "";
+    }
+    return `(function () {\n// PLUGINS CONFIG\n${configsJS}\n${pluginsJS}\n})();\n`;
```

Comment:
typos between `pluginJS` and `pluginsJS` here or am i confused about what's going on?